### ISSUE TITLE
Cyclic dependency upon clj

### ISSUE THREAD

**Issue by arpunk** [lfex/ltest#34]
In order to test `lfex/clj` one needs to specify `ltest` as a dependency, but currently `ltest` depends on `clj`. Is there a way to skip this check? I'm using rebar3.

**Comment by oubiwann** [comment]
Yeah, I've run across this in various places -- tried to fix them up. I think I've ended up settling on not having ltest as a dep of clj -- the tests for clj being something that developers and contributors would deal with more than anyone else, and they would be most able to handle a custom scenario. 

Maybe I just just do something outside the tool for this ... the ``lfe.config`` file has been used in the past to circumvent cyclic dependencies. A goofy ``make`` target would also "work" ...

Anyway, I'll remove the dep from clj for starters -- let me know how that works for you.

(This also makes sense, since a good chunk of clj is on the roadmap for inclusion in LFE as a core library ...)

**Comment by oubiwann** [comment]
Huh ... the latest clj doesn't depend upon ltest, so I must had already addressed this. I'll check the versions released (and in the ``rebar.config`` file(s) ...)

**Comment by oubiwann** [comment]
Yeah, just found where the dep was removed (for this very problem):
 * https://github.com/lfex/clj/commit/c0ae0a22cb4b9a382895320ddfc0edbd931c49ab#diff-31d7a50c99c265ca2793c20961b60979

That was Apr 20, 2015 ... version 0.3.0 of clj was released on Oct 2015. Hrm, and ltest depende upon 0.3.0.

I wonder -- maybe you have an outdated version of clj in your ~/.lfe directory somewhere?

**Comment by arpunk** [comment]
Hey @oubiwann, just saw this. Sorry for that.

No, I do not have a `~/.lfe` dir, however let me try again with fresh deps to see if I can replicate the issue.

**Comment by arpunk** [comment]
You are right, `ltest` was removed from `clj` and now it builds and test passes. `clj` test doesn't run because there are no `ltest` macros available. Shall I open an issue in `clj` or do you have plans for it?

**Comment by oubiwann** [comment]
No worries.

I think we're going to add a test profile in the ``rebar.config`` file to allow for tests to be run (using ``rebar3 as test eunit``) while avoiding cyclic deps ... pretty sure there's a ticket for that.

**Comment by yurrriq** [comment]
I'm working on this now and getting everything on Hex. lutil and ltest depended on clj unnecessarily.

**Comment by yurrriq** [comment]
Fixed and on Hex https://hex.pm/packages/ltest

**Comment by yurrriq** [comment]
Note that the package is from my repo atm, but it should be fine. I wanted to sort out the cyclic dependencies and get some of the Hex-related legwork out of the way. We can always publish new versions. I've got working packages published for:

- [lfe](https://hex.pm/packages/lfe)
- [rebar3_lfe_compile](https://hex.pm/packages/rebar3_lfe_compile)
- [lutil](https://hex.pm/packages/lutil)
- [lodox](https://hex.pm/packages/lodox)
- [kla](https://hex.pm/packages/kla)
- [clj](https://hex.pm/packages/clj)
- [levaindoc](https://hex.pm/packages/levaindoc)
- [rebar3_lfe_version](https://hex.pm/packages/rebar3_lfe_version)
- [ltest](https://hex.pm/packages/ltest)
- [exemplar](https://hex.pm/packages/exemplar)

**Comment by yurrriq** [comment]
This should be fixed now. Can you try it out and let us know, @arpunk?

**Comment by yurrriq** [comment]
ping

**Comment by arpunk** [comment]
@yurrriq I'm using `ltest` latest with rebar3 so I guess this is not an issue anymore? I'm also using LFE v1.1.1

**Comment by yurrriq** [comment]
:+1:

**Comment by oubiwann** [comment]
Okay, I'll close ... we can re-open (or create a new ticket) if it crops up again.
